scripts/ci-e2e.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+cd "${REPO_ROOT}" || exit 1
+
+# shellcheck source=../hack/ensure-go.sh
+source "${REPO_ROOT}/hack/ensure-go.sh"
+# shellcheck source=hack/ensure-kind.sh
+source "${REPO_ROOT}/hack/ensure-kind.sh"
+# shellcheck source=hack/ensure-kubectl.sh
+source "${REPO_ROOT}/hack/ensure-kubectl.sh"
```

Comment:
do we need these? because things are running fine with the `test-sanity` without these, hope we are going to use same docker image for this job too

Reply:
Yes, the same docker image is used for this test. From our previous internal e2e runs, the flow required these binaries to be present.

Reply:
but https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/scripts/ci-smoke-test.sh doesn't have these entires there, then how it is working fine for that script?

Reply:
yes, we don't need kind and kubectl. Only kustomize is required. Probably got confused with another flow. Made the changes.